packages/ui/components/ui/balance-value-view.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { describe, expect, test, vi } from 'vitest';
+import { fireEvent, render, screen } from '@testing-library/react';
+import {
+  Metadata,
+  ValueView,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+import { BalanceValueView } from './balance-value-view';
+import { base64ToUint8Array } from '@penumbra-zone/types/base64';
+
+// Mocking the WalletIcon component
+vi.mock('./icons/wallet', () => ({
+  WalletIcon: ({ className }: { className: string }) => (
+    <div data-testid='wallet-icon' className={className}></div>
+  ),
+}));
+
+describe('<BalanceValueView />', () => {
+  const penumbraMetadata = new Metadata({
+    base: 'upenumbra',
+    display: 'penumbra',
+    symbol: 'UM',
+    penumbraAssetId: {
+      inner: base64ToUint8Array('KeqcLzNx9qSH5+lcJHBB9KNW+YPrBk5dKzvPMiypahA='),
+    },
+    images: [
+      {
+        png: 'https://raw.githubusercontent.com/penumbra-zone/web/main/apps/minifront/public/favicon.png',
+      },
+    ],
+    denomUnits: [
+      {
+        denom: 'penumbra',
+        exponent: 6,
+      },
+      {
+        denom: 'mpenumbra',
+        exponent: 3,
+      },
+      {
+        denom: 'upenumbra',
+        exponent: 0,
+      },
+    ],
+  });
+
+  const valueView = new ValueView({
+    valueView: {
+      case: 'knownAssetId',
+      value: {
+        amount: {
+          hi: 0n,
+          lo: 123_456_789n,
+        },
+        metadata: penumbraMetadata,
+      },
+    },
+  });
+
+  test('renders value and wallet icon without cursor-pointer', () => {
+    const { container } = render(<BalanceValueView valueView={valueView} />);
+
+    expect(container).toHaveTextContent(`123.456789UM`);
+    expect(screen.getByTestId('wallet-icon')).toBeInTheDocument();
+    const clickableElement = screen.getByTestId('wallet-icon').parentElement;
+    expect(clickableElement).not.toHaveClass('cursor-pointer');
+  });
+
+  test('renders with cursor-pointer class and calls onClick when clicked', () => {
```

Comment:
I generally think it's a good idea to split out tests into the single "units" they're testing — i.e., one test that asserts it has the `cursor-pointer` class when `onClick` is passed, and another that asserts that `onClick` is called on click. That gives you better visibility into what's breaking in your unit tests, if something fails. (For example, let's say _both_ the `cursor-pointer` class is missing _and_ the `onClick` handler is not working. This test will fail, but you'll only see the failed expectation for the `cursor-pointer` class, and you won't immediately see that `onClick` isn't working either.)